sematic/plugins/storage/local_storage.py

```diff
@@ -39,20 +37,34 @@ def get_author() -> str:
     def get_version() -> PluginVersion:
         return _PLUGIN_VERSION
 
-    def get_write_location(self, namespace: str, key: str) -> str:
-        return f"/upload/{namespace}/{key}"
+    def get_write_location(
+        self, namespace: str, key: str, user: Optional[User]
+    ) -> Location:
 
-    def get_read_payload(self, namespace: str, key: str) -> ReadPayload:
-        try:
-            with open(os.path.join(_get_data_dir(), namespace, key), "rb") as file:
-                content = file.read()
-        except FileNotFoundError:
-            raise NoSuchStorageKeyError(self.__class__, key)
+        return Location(
+            location=f"{get_api_address()}/api/v1/uploads/{namespace}/{key}/local",
+            headers=_make_headers(user),
+        )
 
-        return ReadPayload(type_=PayloadType.BYTES, content=content)
+    def get_read_location(
+        self, namespace: str, key: str, user: Optional[User]
+    ) -> Location:
+        return Location(
+            location=f"{get_api_address()}/api/v1/uploads/{namespace}/{key}/local",
+            headers=_make_headers(user),
+        )
 
 
-@sematic_api.route("/api/v1/upload/<namespace>/<key>", methods=["PUT"])
+def _make_headers(user: Optional[User]) -> Dict[str, str]:
+    headers = {"Content-Type": "application/octet-stream"}
+
+    if user is not None:
+        headers[API_KEY_HEADER] = user.api_key
+
+    return headers
+
+
+@sematic_api.route("/api/v1/uploads/<namespace>/<key>/local", methods=["PUT"])
```

Comment:
Can you add a comment at the top of `uploads.py` that says that individual storage plugins may add their own endpoints under the `/api/v1/uploads` path? Otherwise people might go to `uploads.py` looking for this endpoint and get confused.